auth/user.go

```diff
@@ -183,6 +183,98 @@ func (user *userImpl) RoleHistory() TimedSetHistory {
 	return user.RoleHistory_
 }
 
+func (user *userImpl) RevokedChannels(since uint64) map[string]uint64 {
+
+	// Grab all channels user has access to
+	accessibleChannels := user.InheritedChannels()
+
+	// Get revoked roles
+	rolesToRevoke := map[string]uint64{}
+	roleHistory := user.RoleHistory()
+	for roleName, history := range roleHistory {
+		if !user.RoleNames().Contains(roleName) {
+			for _, entry := range history.Entries {
+				if entry.StartSeq <= since && entry.EndSeq >= since {
+					mostRecentEndSeq := history.Entries[len(history.Entries)-1]
+					rolesToRevoke[roleName] = mostRecentEndSeq.EndSeq
+				}
+			}
+		}
+	}
+
+	// Store revoked channels to return
```

Comment:
Are both of these inline functions strictly necessary?  For addToCombined, it seems like this would be cleaner as a custom type for the map[string]uint64, with a function like "add(string, uint64)" (that did the check addToCombined is doing).

For revokeChannelHistoryProcessing, what is the benefit of using as an inline function instead of a standalone function?

Reply:
I like the idea for the custom type, that works.
revokeChannelHistoryProcessing - benefit was just its only used internally for that function so seems unnecessary to have it 'bloat' externally but I can do

Reply:
Note to myself: Addressed first part, second part not yet addressed. Can maybe discuss on Monday along with other.